contracts/contracts/ramps/wise/WiseRamp.sol

```diff
@@ -0,0 +1,872 @@
+     * their global account mapping, and deleting the intentHash from the deposit's intentHashes array.
+     */
+    function _pruneIntent(Deposit storage _deposit, bytes32 _intentHash) internal {
+        Intent memory intent = intents[_intentHash];
+
+        delete globalAccount[accounts[intent.onRamper].accountId].currentIntentHash;
+        delete intents[_intentHash];
+        _deposit.intentHashes.removeStorage(_intentHash);
+
+        emit IntentPruned(_intentHash, intent.deposit);
+    }
+
+    /**
+     * @notice Removes a deposit if no outstanding intents AND no remaining deposits. Deleting a deposit deletes it from the
+     * deposits mapping and removes tracking it in the user's accounts mapping.
+     */
+    function _closeDepositIfNecessary(uint256 _depositId, Deposit storage _deposit) internal {
+        uint256 openDepositAmount = _deposit.outstandingIntentAmount + _deposit.remainingDeposits;
+        if (openDepositAmount == 0) {
+            accounts[_deposit.depositor].deposits.removeStorage(_depositId);
+            emit DepositClosed(_depositId, _deposit.depositor);
+            delete deposits[_depositId];
+        }
+    }
+
+    /**
+     * @notice Checks if sustainability fee has been defined, if so sends fee to the fee recipient and intent amount minus fee
+     * to the on-ramper. If sustainability fee is undefined then full intent amount is transferred to on-ramper.
+     */
+    function _transferFunds(bytes32 _intentHash, Intent memory _intent) internal {
+        uint256 fee;
+        if (sustainabilityFee != 0) {
+            fee = (_intent.amount * sustainabilityFee) / PRECISE_UNIT;
+            usdc.transfer(sustainabilityFeeRecipient, fee);
+        }
+
+        uint256 onRampAmount = _intent.amount - fee;
+        usdc.transfer(_intent.to, onRampAmount);
+
+        emit IntentFulfilled(_intentHash, _intent.deposit, _intent.onRamper, _intent.to, onRampAmount, fee);
+    }
+
+    /**
+     * @notice Validate send payment email and check that it hasn't already been used (done on SendProcessor).
+     * Additionally, we validate that the offRamperId matches the one from the specified intent and that enough
+     * was paid off-chain inclusive of the conversionRate.
+     */
+    function _verifyOnRampProof(
+        IWiseSendProcessor.SendData calldata _data,
+        bytes calldata _verifierSignature,
+        bytes32 _intentHash
+    )
+        internal
+        returns(Intent storage intent, Deposit storage deposit)
+    {
+        intent = intents[_intentHash];
+        deposit = deposits[intent.deposit];
+
+        (
+            uint256 amount,
+            uint256 timestamp,
+            bytes32 offRamperId,
+            bytes32 onRamperId,
+            bytes32 intentHash,
+            bytes32 currencyId
+        ) = sendProcessor.processProof(
+            IWiseSendProcessor.SendProof({
+                public_values: _data,
+                expectedTLSParams: deposit.tlsParams,
```

Comment:
Why is this required only for the send processor and not for the registration processors?